### ISSUE TITLE
Asymptote aborts with 'write to pipe failed' when encountering a label when render command is run by abp

### ISSUE THREAD

**Issue by rolandog** [CDSoft/abp#5]
### Issue

I've run into a bug that I don't know how to properly debug; I'll try to document as much as possible.

What happens is that Asymptote crashes when there is a label in the code when run from `abp`, but not when run from the command line.

I tested the following scenarios:

<details>
<summary>Inline code to be parsed:</summary>

    ```{.asy render="{{asy}}" img="img/asy_test_inline" out="img"}
    settings.outformat="svg";
    unitsize(1cm);
    import graph;
    // function to draw
    real f(real x) {
        return sqrt(x);
    }
    // drawing range
    real xmin = 0.0;
    real xmax = 1.0;
    real ymin = f(xmin);
    real ymax = f(xmax);
    // path for f function in drawing range
    path fdraw = graph(f,xmin,xmax,
        operator ..);
    draw(fdraw);
    // x-axis and y-axis configuration
    arrowbar axisarrow = Arrow(TeXHead);
    draw((-0.1,ymin) -- (xmax,ymin), arrow=axisarrow);
    draw((xmin,-0.1) -- (xmin,ymax), arrow=axisarrow);
    // uncomment labels to trigger error
    //label("$x$", (xmax,ymin), align=E);
    //label("$y$", (xmin,ymax), align=N);
    //label("$y=\sqrt{x}$", (xmax,ymax), align=NE);
    ```
</details>

<details>
<summary>Importing `asy_test.asy`:</summary>

In the document:

    ```{.asy render="{{asy}}" img="img/asy_test_inline" out="img" include="src/asy_test.asy"}
    ```

And in the `.asy` file:

```asy
settings.outformat="svg";
unitsize(1cm);
import graph;
// function to draw
real f(real x) {
    return sqrt(x);
}
// drawing range
real xmin = 0.0;
real xmax = 1.0;
real ymin = f(xmin);
real ymax = f(xmax);
// path for f function in drawing range
path fdraw = graph(f,xmin,xmax,
    operator ..);
draw(fdraw);
// x-axis and y-axis configuration
arrowbar axisarrow = Arrow(TeXHead);
draw((-0.1,ymin) -- (xmax,ymin), arrow=axisarrow);
draw((xmin,-0.1) -- (xmin,ymax), arrow=axisarrow);
// uncomment labels to trigger error
//label("$x$", (xmax,ymin), align=E);
//label("$y$", (xmin,ymax), align=N);
//label("$y=\sqrt{x}$", (xmax,ymax), align=NE);
```
</details>
 

### Error Message

I'm using a Makefile to pre-process and generate the documents. I'll try to individually run the generated commands tomorrow, but for now, I'll be pasting these logs (when running with the uncommented labels):

<details>
<summary>`make` output</summary>

```
user@computer:~$ make all
Diagram failed:
settings.outformat="svg";
unitsize(1cm);
import graph;
// function to draw
real f(real x) {
    return sqrt(x);
}
// drawing range
real xmin = 0.0;
real xmax = 1.0;
real ymin = f(xmin);
real ymax = f(xmax);
// path for f function in drawing range
path fdraw = graph(f,xmin,xmax,
    operator ..);
draw(fdraw);
// x-axis and y-axis configuration
arrowbar axisarrow = Arrow(TeXHead);
draw((-0.1,ymin) -- (xmax,ymin), arrow=axisarrow);
draw((xmin,-0.1) -- (xmin,ymax), arrow=axisarrow);
// uncomment labels to trigger error
label("$x$", (xmax,ymin), align=E);
label("$y$", (xmin,ymax), align=N);
label("$y=\sqrt{x}$", (xmax,ymax), align=NE);
Errors:
/usr/share/asymptote/plain_Label.asy: 321.15: abort: write to pipe failed

Error running filter abp:
Filter returned error status 1
make: *** [Makefile:351: mwe-abp-asymptote-label-bug.html.md] Error 83
```
</details>
 

### Running Asymptote from the command-line

And here's the output I get when running Asymptote from the command-line:

```
user@computer:~$ asy -v -f svg -o asy_test.svg asy_test.asy
Processing asy_test
Wrote asy_test.svg
```

I don't encounter any errors when running from the cli.

### Environment

<details>
<summary>Environment info details</summary>

My `abp` version is `0.1.8.7`:
```
user@computer:~$ abp -v
abp 0.1.8.7
Abstract Processor for Pandoc

Copyright (C) 2019-2020 Christophe Delord
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Released under the GPL-3 license.

See http://cdelord.fr/abp for further information.
```

My OS is Ubuntu Focal Fossa (20.04):
```
user@computer:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04 LTS
Release:	20.04
Codename:	focal
```

Also, I'm using gnome-terminal and bash:
```
user@computer:~$ ps -p $$ -o args,ppid
COMMAND                        PPID
bash                           1234
user@computer:~$ ps -p 1234 -o args
COMMAND
/usr/libexec/gnome-terminal-server
```
</details>

**Comment by CDSoft** [comment]
I have a similar behavior on Fedora. Sometimes I get the same error, sometimes asy runs forever.
It seems that asy/latex uses stdin with is not redirected to a pipe when the process is created.
If I explicitly redirect stdin to a pipe it does not crash but I'm not sure to understand everything...
I'll push a fix soon.

**Comment by CDSoft** [comment]
Should be fixed in 0.1.8.8 (tested on Fedora 32).

**Comment by rolandog** [comment]
Awesome! Thanks so much! It's now working like a charm! I'll mark this as closed!
